src/main/scala/Destination.scala

```diff
@@ -17,8 +19,24 @@ case object IntListDestination extends Destination[List[Int]] {
   def transform(input: List[Int]): List[Int] = input.reverse
 }
 
-case object JsonDestination extends Destination[String] {
-  def save(filename: String, data: String): Unit =
-    new PrintWriter(s"./src/main/resources/output/$filename") { write(data); close }
-  def transform(input: String): String = input.toLowerCase
+case object JsonDestination extends Destination[Json] {
+  def save(filename: String, data: Json): Unit =
+    new PrintWriter(s"./src/main/resources/output/$filename") { write(data.toString); close }
+
+  def transform(input: Json): Json = {
+    implicit val decodeOwner: Decoder[Owner] = Decoder.forProduct2("first_name", "last_name")(Owner.apply)
+    implicit val decodePet: Decoder[Pet] = Decoder.forProduct2("name", "pet_type")(Pet.apply)
+    implicit val decodePetOwner: Decoder[PetOwner] = Decoder.forProduct4("id", "owner", "email", "pet")(PetOwner.apply)
```

Comment:
`implicit val decodeOwner: Decoder[Owner] = deriveDecoder[PetOwner]` ([example](https://circe.github.io/circe/codecs/semiauto-derivation.html)) was not working for any of the implicit decoders or encoders. 

Thought _maybe_ it could be due to problems with imports...?